tests/utilTest.cpp

```diff
@@ -0,0 +1,421 @@
+    std::vector<unsigned char> bytes(8);
+
+    val = 0ULL;
+    expected = { 0, 0, 0, 0, 0, 0, 0, 0 };
+    libzerocash::convertIntToBytesVector(val, bytes);
+    BOOST_CHECK( expected == bytes );
+
+    val = 1ULL;
+    expected = { 0, 0, 0, 0, 0, 0, 0, 1 };
+    libzerocash::convertIntToBytesVector(val, bytes);
+    BOOST_CHECK( expected == bytes );
+
+    val = 0xffffffffffffffffULL;
+    expected = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
+    libzerocash::convertIntToBytesVector(val, bytes);
+    BOOST_CHECK( expected == bytes );
+
+    val = 0x8000000080000001ULL; // sign extension
+    expected = { 0x80, 0x00, 0x00, 0x00, 0x80, 0x00, 0x00, 0x01 };
+    libzerocash::convertIntToBytesVector(val, bytes);
+    BOOST_CHECK( expected == bytes );
+
+    // The following two tests aren't necessarily desired results. They merely
+    // document the behavior so that we'll be alerted if it changes in the
+    // future.
+
+    val = 0xffffffffdeadbeefULL; // truncation
+    expected = { 0xde, 0xad, 0xbe, 0xef };
+    std::vector<unsigned char> small_bytes(4);
+    libzerocash::convertIntToBytesVector(val, small_bytes);
+    BOOST_CHECK( expected == small_bytes );
+
+    val = 0xf1f2f3f401020304ULL; // bytes buffer is too big
+    // The first byte is 4 because `>> 64` is undefined, and that's the result
+    // it has on my system (note that it's the same as the original LSB).
+    expected = { 0x04, 0xf1, 0xf2, 0xf3, 0xf4, 0x01, 0x02, 0x03, 0x04 };
+    std::vector<unsigned char> big_bytes(9);
+    libzerocash::convertIntToBytesVector(val, big_bytes);
+    BOOST_CHECK( expected == big_bytes);
+}
+
+BOOST_AUTO_TEST_CASE( testConvertBytesVectorToInt ) {
+    uint64_t val;
+    uint64_t expected;
+    std::vector<unsigned char> bytes;
+
+    bytes = { 0, 0, 0, 0, 0, 0, 0, 0 };
+    expected = 0ULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+
+    bytes = { 0, 0, 0, 0, 0, 0, 0, 1 };
+    expected = 1ULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+
+    bytes = { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff };
+    expected = 0xffffffffffffffffULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+
+    bytes = { 0x80, 0x00, 0x00, 0x00, 0x80, 0x00, 0x00, 0x01 };
+    expected = 0x8000000080000001ULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+
+    bytes = { 0xde, 0xad, 0xbe, 0xef }; // opposite of truncation
+    expected = 0xdeadbeefULL;
+    val = libzerocash::convertBytesVectorToInt(bytes);
+    BOOST_CHECK( expected == val );
+}
+
+BOOST_AUTO_TEST_CASE( testConcatenateTwoBoolVectors ) {
+    std::vector<bool> front = { 0, 1, 0 };
+    std::vector<bool> back = { 1, 0, 1 };
+    std::vector<bool> expected = { 0, 1, 0, 1, 0, 1 };
+    std::vector<bool> actual;
+    libzerocash::concatenateVectors(front, back, actual);
```

Comment:
Test LGTM.

hrm...  It looks like this API does allocation into the output.  If we're going to have the library do allocation, then why not return a newly allocated output vector?  If we're not going to do allocation in the library, then let's be consistent about it.  If some functions allocate and some do not, let's clarify that in their names and/or type signatures.